### ISSUE TITLE
Linter objects to "any" even in type guards

### ISSUE THREAD

**Issue by lcrocker** [denoland/deno_lint#628]
The linter (correctly, in general) objects to any use of "any", but there's one place where TypeScript all but requires it, which is user-defined type guards. For example:

```ts
function isMyType(obj: any) obj is MyType {
    if (! obj) return false;
    if (! (obj instanceof Object)) return false;
    if (! ('number' === typeof obj.count)) return false;
    return true;
}
```

One might argue that this is an issue with TypeScript in that it doesn't allow the use of "unknown" here without throwing errors and doesn't have any other way to check for the existence and type of properties of an unknown. But until this is fixed in TypeScript, it would be handy if the linter made an exception for type guards.

**Comment by kitsonk** [comment]
Correct, `unknown` should be used in user-defined type guards.

**Comment by lcrocker** [comment]
Using "unknown" is user-defined type guards is the right thing to do, but it doesn't work, at least in the current state of TypeScript because there's no way to assert the presence of a property.
See https://github.com/microsoft/TypeScript/issues/25720 . We should say explicitly that we won't fix this because we expect TypeScript to fix their end.

**Comment by Soremwar** [comment]
@lcrocker 

This is a safe way of asserting a value using unknown, it won't trigger getters or return false on validation over falsy values

```ts
function checkProperty(value: unknown){
  if(typeof value === "object" && value !== null){
    if("my_property" in value){
      return true;
    }
  }
  return false;
}
```
